### PR TITLE
Fixes for data-handling in NSURLSession implemenation especially re errors

### PR DESCRIPTION
**Pull request by boxtcarpel**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @boxtcarpel has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m [hunk 1]
```diff
@@ -555,7 +555,17 @@ - (void)finishURLSessionTaskWithData:(NSData *)data response:(NSURLResponse *)re
 
 - (void)sessionTask:(NSURLSessionTask *)sessionTask didFinishWithResponse:(NSURLResponse *)response error:(NSError *)error
 {
-    [self finishURLSessionTaskWithData:nil response:response error:error];
+    [self finishURLSessionTaskWithData:self.responseData response:response error:error];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m:558]: Note these pretty significant behavior changes here.

File: BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m [hunk 2]
```diff
@@ -555,7 +555,17 @@ - (void)finishURLSessionTaskWithData:(NSData *)data response:(NSURLResponse *)re
...
+}
+
+- (void)processIntermediateResponse:(NSURLResponse *)response
```
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m:561]: I think we could remove this delegate method processIntermediateResponse completely together with BOXURLSessionManager's implementation of delegate method URLSession:dataTask:didReceiveResponse:completionHandler: given that we have no need to intercept this method for the ability that it provides, and we still have access to response at the last callback which exists for both background/non-background URLSession:task:didCompleteWithError:, this also helps avoid processing response twice.

Comment (boxtcarpel) [reply BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m:561]: Sometimes you need the response first... We might need to solve this another way

**Comment by thuyBox** [comment]
+2, we'll follow-up with avoiding processing response twice for fore-ground download/upload

**Pull request merged**